chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -2127,14 +2133,92 @@ impl PeerManagerActor {
     }
 }
 
+/// Fetches NetworkInfo, which contains a bunch of stats about the
+/// P2P network state. Currently used only in tests.
+/// TODO(gprusak): In prod, NetworkInfo is pushed periodically from PeerManagerActor to ClientActor.
+/// It would be cleaner to replace the push loop in PeerManagerActor with a pull loop
+/// in the ClientActor.
+impl Handler<GetNetworkInfo> for PeerManagerActor {
+    type Result = NetworkInfo;
+    fn handle(&mut self, _: GetNetworkInfo, _ctx: &mut Self::Context) -> NetworkInfo {
+        self.get_network_info()
+    }
+}
+
 impl Handler<SetChainInfo> for PeerManagerActor {
     type Result = ();
     fn handle(&mut self, info: SetChainInfo, _ctx: &mut Self::Context) {
-        if self.state.accounts_data.set_keys(info.0.tier1_accounts.clone()) {
-            // TODO(gprusak): the set of tier1 accounts has changed.
-            // We might miss some data, so start a full sync with connected peers.
-        }
-        *self.state.chain_info.write() = info.0;
+        let now = self.clock.now_utc();
+        let info = info.0;
```

Comment:
```suggestion
        let SetChainInfo(info) = info;
```

would be a bit clener way to write this -- won't left the reader wondering what *else* is there besides `.0`

Reply:
done